lib/beaker-pe/pe-client-tools/executable_helper.rb

```diff
@@ -0,0 +1,153 @@
+module Beaker
+  module DSL
+    module PEClientTools
+      module ExecutableHelper
+
+        # puppet-access helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-access command
+        # @param [String] args The arguments to puppet-access
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_access_on(*args, &block)
+          Private.new.tool(:access, *args, &block)
+        end
+
+        # puppet-code helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-code command
+        # @param [String] args The arguments to puppet-code
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_code_on(*args, &block)
+          Private.new.tool(:code, *args, &block)
+        end
+
+        # puppet-job helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-job command
+        # @param [String] args The arguments to puppet-job
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_job_on(*args, &block)
+          Private.new.tool(:job, *args, &block)
+        end
+
+        # puppet-app helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-app command
+        # @param [String] args The arguments to puppet-app
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_app_on(*args, &block)
+          Private.new.tool(:app, *args, &block)
+        end
+
+        # puppet-db helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-db command
+        # @param [String] args The arguments to puppet-db
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_db_on(*args, &block)
+          Private.new.tool(:db, *args, &block)
+        end
+
+        # puppet-query helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-query command
+        # @param [String] args The arguments to puppet-query
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_query_on(*args, &block)
```

Comment:
@er0ck on Windows we will always need to test the discrete executable.  The subcommand
```
puppet job run 
```
will never be on the path.